packages/alpinejs/src/directives.js

```diff
@@ -15,7 +15,22 @@ export function setPrefix(newPrefix) {
 
 let directiveHandlers = {}
 
-export function directive(name, callback) {
+export function directive(nameOrOptions, callback) {
+    let name = nameOrOptions
+    if (nameOrOptions.name){
+        name = nameOrOptions.name
+        let order = nameOrOptions.order
+        if (order){
+            if (!Number.isInteger(order)){
+                try{
```

Comment:
Idea is great, I do remember something about this coming out even at Alpine Day when v3 was announced.

But I think this is needlessly complicated way to write this.

Why use a function to to...what exactly?

If it's not a number, just discard it and maybe `console.warn` about it. No need for extra steps that don't, to my eyes, add any value. Or is the point to allow someone to pay anything that resolves to a number? but why can't the end user just resolve the number themselves?

Reply:
`order` could be a number or an expression, like "`orders.data - 1`". Since there are no exported const or enums for the internal `directiveOrder` definition, you have to dive into the source code to find out the order value of `data`
```javascript

let directiveOrder = {
    'ignore'        : 100,
    'ref'           : 200,
    'data'          : 300,
    'id'            : 400,
    'bind'          : 500,
    'init'          : 600,
    'for'           : 700,
    'mask'          : 800,
    'model'         : 900,
    'modelable'     : 1000,
    'transition'    : 1100,
    'show'          : 1200,
    'if'            : 1300,
    // DEFAULT      : DEFAULT_ORDER = 10000
    'teleport'      : 20000,
}
```
With expression supported, if you want to insert a directive with higher priority than `bind`, just give an expression as "`orders.bind - 1`". Lower priority than `model`? Just use "orders.model + 1".  You do not have to know what's the exact order value of target directive.

Reply:
Ah, okay, very smart!!!

Had not thought of that kind of use case.